Source/ASTextNode2.mm

```diff
@@ -48,11 +54,11 @@ @implementation ASTextCacheValue
  * NOTE: Be careful to copy `text` if needed.
  */
 static NS_RETURNS_RETAINED ASTextLayout *ASTextNodeCompatibleLayoutWithContainerAndText(ASTextContainer *container, NSAttributedString *text)  {
-  static dispatch_once_t onceToken;
-  static AS::Mutex *layoutCacheLock;
+  // Allocate layoutCacheLock on the heap to prevent destruction at app exit (https://github.com/TextureGroup/Texture/issues/136)
+  static auto *layoutCacheLock = new AS::Mutex;
   static NSCache<NSAttributedString *, ASTextCacheValue *> *textLayoutCache;
+  static dispatch_once_t onceToken;
```

Comment:
This is overall missing several of the google changes, like for text attachments and highlighting. Is this a subset of the [Unified PR](https://github.com/TextureGroup/Texture/pull/1986) we put up, or does it include additional changes? Do you think it would be easiest as far as integrating the Google work to essentially break that PR into smaller topical pieces, perhaps with modifications?

Reply:
> We haven't observed the issue described here #136 but it's not clear how it would come up or how this change corrects it. Can you clarify?

I'm actually confused as to where this comment came from. I don't see it in master's [ASTN2](https://github.com/TextureGroup/Texture/blob/master/Source/ASTextNode2.mm#L53)